lib/custom/src/MShop/Service/Provider/Payment/Stripe.php

```diff
@@ -33,20 +111,276 @@ protected function getConfigPrefix()
 	 */
-	protected function getValue( $key, $default = null )
+	public function getConfigBE()
 	{
-		switch( $key )
+		$list = [];
+
+		$prefix = $this->getConfigPrefix();
+		$config = $this->beConfig;
+
+		if( $prefix !== 'omnipay' ) {
+			unset( $config['type'], $config['onsite'] );
+		}
+
+		foreach( $config as $key => $config )
 		{
-			case 'type': return 'Stripe';
-			case 'onsite': return true;
+			$config['code'] = $prefix . '.' . $config['code'];
+			$list[$prefix.'.'.$key] = new \Aimeos\MW\Criteria\Attribute\Standard( $config );
 		}
 
-		return parent::getValue( $key, $default );
+		return $list;
 	}
+
+
+	/**
+	 * Checks the backend configuration attributes for validity.
+	 *
+	 * @param array $attributes Attributes added by the shop owner in the administraton interface
+	 * @return array An array with the attribute keys as key and an error message as values for all attributes that are
+	 * 	known by the provider but aren't valid
+	 */
+	public function checkConfigBE( array $attributes )
+	{
+		$errors = parent::checkConfigBE( $attributes );
+
+		$prefix = $this->getConfigPrefix();
+		$config = $this->beConfig;
+		$list = [];
+
+		if( $prefix !== 'omnipay' ) {
+			unset( $config['type'], $config['onsite'] );
+		}
+
+		foreach( $config as $key => $cfg )
+		{
+			$cfg['code'] = $prefix . '.' . $cfg['code'];
+			$list[$prefix.'.'.$key] = $cfg;
+		}
+
+		return array_merge( $errors, $this->checkConfig( $list, $attributes ) );
+	}
+
+	/**
+	 * Returns the configuration attribute definitions of the provider to generate a list of available fields and
+	 * rules for the value of each field in the frontend.
+	 *
+	 * @param \Aimeos\MShop\Order\Item\Base\Iface $basket Basket object
+	 * @return array List of attribute definitions implementing \Aimeos\MW\Common\Critera\Attribute\Iface
+	 */
+	public function getConfigFE(\Aimeos\MShop\Order\Item\Base\Iface $basket)
+	{
+		return [];
+	}
+
+	/**
+	 * Returns the data passed to the Omnipay library
+	 *
+	 * @param \Aimeos\MShop\Order\Item\Base\Iface $base Basket object
+	 * @param $orderid Unique order ID
+	 * @param array $params Request parameter if available
+	 */
+	protected function getData( \Aimeos\MShop\Order\Item\Base\Iface $base, $orderid, array $params )
+	{
+		$data = parent::getData($base,$orderid,$params);
+		if( isset($params['paymenttoken']) ){
+			$data['token'] = $params['paymenttoken'];
+		}
+		return $data;
+	}
+
+	/**
+	 * Checks the frontend configuration attributes for validity.
+	 *
+	 * @param array $attributes Attributes entered by the customer during the checkout process
+	 * @return array An array with the attribute keys as key and an error message as values for all attributes that are
+	 *    known by the provider but aren't valid resp. null for attributes whose values are OK
+	 */
+	public function checkConfigFE(array $attributes)
+	{
+		return [];
+		//return $this->checkConfig($this->feConfig, $attributes);
+	}
+
+	public function getHTMLform()
```

Comment:
Can you name the method according to the camel case standard, i.e. getHtmlForm()?

Reply:
Done